shared/libraries/websocket_streaming/src/streaming_client.cpp

```diff
@@ -213,7 +213,17 @@ void StreamingClient::onProtocolMeta(daq::streaming_protocol::ProtocolHandler& p
 
                 std::promise<void> signalInitPromise;
                 std::future<void> signalInitFuture = signalInitPromise.get_future();
-                signalInitializedStatus.insert({signalId, std::make_pair(std::move(signalInitPromise), std::move(signalInitFuture))});
+                signalInitializedStatus.insert_or_assign(signalId, std::make_pair(std::move(signalInitPromise), std::move(signalInitFuture)));
+
+                if (auto signalIt = signals.find(signalId); signalIt == signals.end())
+                {
+                    auto inputSignal = std::make_shared<InputSignal>();
+                    signals.insert({signalId, inputSignal});
+                }
+                else
+                {
+                    LOG_W("Signal id {} is already known as available", signalId);
```

Comment:
I'd try to log more descriptive messages in such cases. Something similar to "Received duplicate signal ID. Only one the first received signal will be streamed.".